service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseRoleLookupService.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.CaseAccessDataStoreApi;
+import uk.gov.hmcts.reform.ccd.model.CaseAssignedUserRole;
+import uk.gov.hmcts.reform.ccd.model.CaseAssignedUserRolesResource;
+import uk.gov.hmcts.reform.fpl.enums.SolicitorRole;
+import uk.gov.hmcts.reform.fpl.request.RequestData;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class CaseRoleLookupService {
+
+    private final CaseAccessDataStoreApi api;
+    private final RequestData requestData;
+    private final AuthTokenGenerator authTokenGenerator;
+
+    public List<SolicitorRole> getCaseSolicitorRolesForCurrentUser(String caseId) {
```

Comment:
I think joe did something vv similar in https://github.com/hmcts/fpl-ccd-configuration/pull/2919, would be good to be consistent and have only one way of extracting what is required

Reply:
I couldn't find it in this PR. Can you point me to it, please?

Reply:
Check the user service

Reply:
Got you.
Will move my method (and its test) to the UserService.

Reply:
Actually, now that I started moving things around, I've realised these two classes have a crucial difference. One is about IDAM roles and the other is about case roles.
I think I'll keep them apart, if you don't mind.

Reply:
the method I was talking about is
```
    public boolean hasAnyCaseRoleFrom(List<CaseRole> caseRoles, String caseId) {
        Set<CaseRole> userRoles = caseAccessService.getUserCaseRoles(caseId);

        return caseRoles.stream().anyMatch(userRoles::contains);
    }
```
which is to do with case roles

Reply:
Oh, I see. I think these achieve different things, actually.
In my opinion, I'd keep them separate.
I'd be happy to have a call about this, though, as I might be missing some angle.